support/db/dbtest/db.go

```diff
@@ -128,8 +128,8 @@ func checkReadOnly(t testing.TB, DSN string) {
 	if !rows.Next() {
 		_, err = tx.Exec("CREATE ROLE user_ro WITH LOGIN PASSWORD 'user_ro';")
 		if err != nil {
-			// Handle race condition by ignoring the error if it's a duplicate key violation
-			if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == "23505" {
+			// Handle race condition by ignoring the error if it's a duplicate key violation or duplicate object error
+			if pqErr, ok := err.(*pq.Error); ok && (pqErr.Code == "23505" || pqErr.Code == "42710") {
 				return
 			}
```

Comment:
we should log pqErr.Code in case there are other cases that we have missed

```suggestion
			} else if ok {
				t.Logf("pq error code: %s", pqErr.Code)
			}
```